File: m2/demos/Personaje.java
public class Personaje {
    private String nombre;
    private int defensa;
    private int poder;
    private int salud;

    // Constructor
    public Personaje(String nombre, int defensa, int poder, int salud) {
        this.nombre = nombre;
        this.defensa = defensa;
        this.poder = poder;
        this.salud = salud;
    }

    public Personaje(String nombre) {
        this.nombre = nombre;
    }

    // Getters y Setters
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getDefensa() {
        return defensa;
    }

    public void setDefensa(int defensa) {
        this.defensa = defensa;
    }

    public int getPoder() {
        return poder;
    }

    public void setPoder(int poder) {
        this.poder = poder;
    }

    public int getSalud() {
        return salud;
    }

    public void setSalud(int salud) {
        this.salud = salud;
    }

    // Método para recibir daño
    public void recibirDanio(int danio) {
        int danioFinal = danio - defensa;
        if (danioFinal > 0) {
            salud -= danioFinal;
            if (salud < 0) {
                salud = 0; // Salud mínima es 0
            }
        }
    }

    // Método para atacar
    public int atacar() {
        return poder;
    }

    // Método para mostrar información del personaje
    @Override
    public String toString() {
        return "Personaje{" +
                "nombre='" + nombre + '\'' +
                ", defensa=" + defensa +
                ", poder=" + poder +
                ", salud=" + salud +
                '}';
    }
}

File: m2/demos/Personaje.cs
public class Personaje
{
    public string Nombre { get; set; }
    public int Defensa { get; set; }
    public int Poder { get; set; }
    public int Salud { get; set; }

    // Constructor
    public Personaje(string nombre, int defensa, int poder, int salud)
    {
        Nombre = nombre;
        Defensa = defensa;
        Poder = poder;
        Salud = salud;
    }

    // Método para recibir daño
    public void RecibirDanio(int danio)
    {
        int danioFinal = danio - Defensa;
        if (danioFinal > 0)
        {
            Salud -= danioFinal;
            if (Salud < 0)
            {
                Salud = 0; // La salud mínima es 0
            }
        }
    }

    // Método para atacar
    public int Atacar()
    {
        return Poder;
    }

    // Método para mostrar información del personaje
    public override string ToString()
    {
        return $"Personaje {{ Nombre: {Nombre}, Defensa: {Defensa}, Poder: {Poder}, Salud: {Salud} }}";
    }
}

// Ejemplo de uso
class Program
{
    static void Main(string[] args)
    {
        Personaje personaje = new Personaje("Guerrero", 10, 20, 100);
        Console.WriteLine(personaje.ToString());
        
        personaje.RecibirDanio(30);
        Console.WriteLine($"Después de recibir daño: {personaje}");
        
        Console.WriteLine($"El personaje ataca con un poder de: {personaje.Atacar()}");
    }
}
